packages/rum-core/src/common/observers/page-clicks.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import { USER_INTERACTION } from '../constants'
+
+export function observePageClicks(transactionService) {
+  const clickHandler = event => {
+    if (event.target instanceof Element) {
```

Comment:
Any reason to pick Element and not `EventTarget` ? Can you add a comment?

Reply:
If we check against `EventTarget` we will need to handle also events that are not corresponding to a DOM element.

Two examples:

1. `window.dispatchEvent(new Event('click'))` will cause the event.target to be an instance of [Window](https://developer.mozilla.org/en-US/docs/Web/API/Window). 
2. `document.dispatchEvent(new Event('click'))` will cause the event.target to be an instance of [Document](https://developer.mozilla.org/en-US/docs/Web/API/Document).

It's important to mention that while I was working on this feature I had a "prototype" where I was handling them. In case of Window I was creating a transaction with the name "click - window" and in case of Document the name was "click - document"

But our goal is to capture **real** user click interactions (the target will be always `Element`). I decided not to handle them.

I'm going to add to the comment explaining the why ✍️ 😄

Reply:
Gotcha, that makes sense. Thanks for the details :)